packages/vm/tests/api/customPrecompiles.spec.ts

```diff
@@ -0,0 +1,120 @@
+import tape from 'tape'
+import VM from '../../src'
+import { Address } from 'ethereumjs-util'
+import { PrecompileInput } from '../../src/evm/precompiles'
+import { ExecResult } from '../../src/evm/evm'
+
+const sender = new Address(Buffer.from('44'.repeat(20), 'hex'))
+const newPrecompile = new Address(Buffer.from('ff'.repeat(20), 'hex'))
+const shaAddress = new Address(Buffer.from('0000000000000000000000000000000000000002', 'hex'))
+const expectedReturn = Buffer.from('1337', 'hex')
+const expectedGas = BigInt(10)
+
+// TODO if `opts` is removed (so function has no arguments) then tests fail!? Why?
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function customPrecompile(opts: PrecompileInput): ExecResult {
+  return {
+    gasUsed: expectedGas,
+    returnValue: expectedReturn,
+  }
+}
+
+tape('VM -> custom precompiles', (t) => {
+  t.test('should override existing precompiles', async (st) => {
+    const VMOverride = new VM({
+      customPrecompiles: [
+        {
+          address: shaAddress,
+          function: customPrecompile,
+        },
+      ],
+    })
+    const result = await VMOverride.runCall({
+      to: shaAddress,
+      gasLimit: BigInt(30000),
+      data: Buffer.from(''),
+      caller: sender,
+    })
+    st.ok(result.execResult.returnValue.equals(expectedReturn), 'return value is correct')
+    st.ok(result.execResult.gasUsed === expectedGas, 'gas used is correct')
```

Comment:
just a note, i'll add a commit to change these, but we can use `st.equal` with bigint 👍 and provides better error messages (expected, result) on error

```suggestion
    st.equal(result.execResult.gasUsed, expectedGas, 'gas used is correct')
```